design/fencing_mitigation.md

```diff
@@ -0,0 +1,36 @@
+# Problem Statement
+
+We need to ensure that only one node can be fenced at a time and that Ceph is fully recovered (has healthy state) before any fencing is initiated. The available pattern for limiting fencing is the MachineDisruptionBudget which allows us to specify maxUnavailable. However, this won’t be sufficient to ensure that Ceph has recovered before fencing is initiated.
+
+Therefore, we will control how many nodes match the MDB by dynamically adding and removing labels as will as dynamically updating the MDB. By manipulating the MDB into a state where desiredHealthy <= currentHealthy, we can disable fencing.
+
+# Design:
+
+## node-controller
+Each `Machine` part of a `StorageCluster` will be labeled with the key `ocs.openshift.io/<cluster-name>.
```

Comment:
Per discussion yesterday, the nodes labeled, `cluster.ocs.openshift.io/<cluster-name>=""` will be OCS nodes. This is configured at install time or later via day2 ops. Having a node w/ the above label means it is eligible to run OCS components, but it does not mean that it actually has any assigned to it.

I assume the machine label mentioned here is different from the node label used to mark OCS nodes. I'm wondering if we can create a more descriptive name for the machine label: `osd.ocs.openshift.io/<cluster-name>=""`

Also, while I'm thinking about it, don't we need the same disruption logic for mon pods?